src/main/java/org/gradle/profiler/GradleClientSpec.java

```diff
@@ -42,19 +43,19 @@ public GradleClient create(GradleBuildConfiguration buildConfiguration, Invocati
     AndroidStudio("Android Studio") {
         @Override
         public GradleClient create(GradleBuildConfiguration buildConfiguration, InvocationSettings invocationSettings) {
-            return new StudioGradleClient(buildConfiguration, invocationSettings, NEVER);
+            return new StudioGradleClient((StudioGradleBuildConfiguration) buildConfiguration, invocationSettings, NEVER);
```

Comment:
I wonder if we can avoid the cast here by introducing new types. I am not sure if it is worth it.

Reply:
It would require some extra refactoring, for example we would most likely have to totally separate StudioGradleScenarioInvoker and GradleScenarioInvoker.

Reply:
I wondered if you could share a common interface for the two enums, so at least `StudioGradleScenarioInvoker` would be a `GradleScenarioInvoker` as well. Anyway, it is probably not worth it at this time to separate the two.